### DISCUSSION TITLE
Permission denied error when runing in Linux via docker

### DISCUSSION THREAD

**Discussion by 23WilliamDing** [hardkoded/puppeteer-sharp#2598] [Q&A]
@undefinedfunction  Hi, puppeteer-sharp works well in Windows, but it cannot work when I deploy my application(.net core 3.1) in Linux via docker, even I refer to some discussions here, it still cannot work, Do you have any suggestions? Thanks! Do I miss something?

![image](https://github.com/hardkoded/puppeteer-sharp/assets/30248071/d4bf294d-3a3b-469c-966e-29f8406fed67)

INFO  2024-04-12 16:34:41,281 [42   ] pNetCore.Cors.Infrastructure.CorsService - Request does not have permission to access the resource.
INFO  2024-04-12 16:34:41,282 [42   ] uthentication.JwtBearer.JwtBearerHandler - Successfully validated the token.
INFO  2024-04-12 16:34:41,283 [42   ] ft.AspNetCore.Routing.EndpointMiddleware - Executing endpoint 'BiNewPlatform.NGS.NGSServices.TestSTRAnalysis (BiNewPlatform.Application)'
INFO  2024-04-12 16:34:41,287 [42   ] c.Infrastructure.ControllerActionInvoker - Route matched with {area = "app", action = "TestSTRAnalysis", controller = "NGSServices"}. Executing controller action with signature System.Threading.Tasks.Task TestSTRAnalysis() on controller BiNewPlatform.NGS.NGSServices (BiNewPlatform.Application).
**ERROR 2024-04-12 16:34:41,342 [37   ] Mvc.ExceptionHandling.AbpExceptionFilter - Permission denied
System.ComponentModel.Win32Exception (13): Permission denied**

**Comment by undefinedfunction** [comment]
I don't remember what I wrote about previously to warrant to be tagged in this.

AFAIK the 13 denotes ERROR_INVALID_DATA and I think that means you're trying to run Linux on Windows or something. I would make sure you are running Docker properly and building a suitable build.

There isn't enough info to make more than a guess

**Comment by kblok** [comment]
On the other hand. If you are using a docker image. It would be way better for you to ship the browser with the docker image so you don't waste time downloading the browser.
